tests/src/Interop/PInvoke/Structures/PInvoke/AsParamNative.h

```diff
@@ -0,0 +1,954 @@
+    {
+        return false;
+    }
+    return true;
+}
+
+struct ByteStructPack2Explicit // size = 2 bytes
+{
+    BYTE b1;
+    BYTE b2;
+};
+
+void PrintByteStructPack2Explicit(ByteStructPack2Explicit* str, char* name)
+{
+    printf("\t%s.b1 = %d", name, str->b1);
+    printf("\t%s.b2 = %d", name, str->b2);
+}
+
+void ChangeByteStructPack2Explicit(ByteStructPack2Explicit* p)
+{
+    p->b1 = 64;
+    p->b2 = 64;
+}
+
+bool IsCorrectByteStructPack2Explicit(ByteStructPack2Explicit* p)
+{
+    if(p->b1 != 32 || p->b2 != 32)
+        return false;
+    return true;
+}
+
+#pragma pack (push)
+#pragma pack (8)
+
+struct ShortStructPack4Explicit // size = 4 bytes
+{
+    SHORT s1;
+    SHORT s2;
+};
+
+#pragma pack (pop)
+
+void PrintShortStructPack4Explicit(ShortStructPack4Explicit* str, char* name)
+{
+    printf("\t%s.s1 = %d", name, str->s1);
+    printf("\t%s.s2 = %d", name, str->s2);
+}
+
+void ChangeShortStructPack4Explicit(ShortStructPack4Explicit* p)
+{
+    p->s1 = 64;
+    p->s2 = 64;
+}
+
+bool IsCorrectShortStructPack4Explicit(ShortStructPack4Explicit* p)
+{
+    if(p->s1 != 32 || p->s2 != 32)
+        return false;
+    return true;
+}
+
+#pragma pack (push)
+#pragma pack (8)
+
+struct IntStructPack8Explicit // size = 8 bytes
+{
+    INT i1;
+    INT i2;
+};
+
+#pragma pack (pop)
+
+void PrintIntStructPack8Explicit(IntStructPack8Explicit* str, char* name)
+{
+    printf("\t%s.i1 = %d", name, str->i1);
+    printf("\t%s.i2 = %d", name, str->i2);
+}
+
+void ChangeIntStructPack8Explicit(IntStructPack8Explicit* p)
+{
+    p->i1 = 64;
+    p->i2 = 64;
+}
+
+bool IsCorrectIntStructPack8Explicit(IntStructPack8Explicit* p)
+{
+    if(p->i1 != 32 || p->i2 != 32)
+        return false;
+    return true;
+}
+
+#pragma pack (push)
+#pragma pack (8)
+
+struct LongStructPack16Explicit // size = 16 bytes
+{
+    long l1;
+    char padding1[4];
+    long l2;
+    char padding2[4];
+};
+
+#pragma pack (pop)
+
+void PrintLongStructPack16Explicit(LongStructPack16Explicit* str, char* name)
+{
+    printf("\t%s.l1 = %ld", name, str->l1);
+    printf("\t%s.l2 = %ld", name, str->l2);
+}
+
+void ChangeLongStructPack16Explicit(LongStructPack16Explicit* p)
+{
+    p->l1 = 64;
+    p->l2 = 64;
+}
+
+bool IsCorrectLongStructPack16Explicit(LongStructPack16Explicit* p)
+{
+    if(p->l1 != 32 || p->l2 != 32)
+        return false;
+    return true;
+}
+
+#pragma pack (push)
+#pragma pack (8)
+
+struct StringExplicit // size = 20 bytes
+{
+    LPCSTR str1;
+#ifndef _WIN64
+    char padding1[4];
+#endif
+    LPCTSTR str2;
+#ifndef _WIN64
+    char padding2[4];
+#endif
+    CHAR charr[10];
+};
+
+#pragma pack (pop)
+
+void PrintStringExplicit(StringExplicit* str, char* name)
+{
+    printf("\t%s.str1 = %s", name, str->str1);
+    printf("\t%s.str2 = %s", name, str->str2);
+    wprintf(L"\t%S.charr = %S", name, str->charr);
+}
+
+void ChangeStringExplicit(StringExplicit* p)
+{
+    LPCSTR str1 = "Managed_LPCSTR\0";
+    LPCTSTR str2 = (LPCTSTR)L"Managed_LPCSTR\0";
```

Comment:
@AaronRobinsonMSFT @luqunl There's warning here, do you have any idea about how to fix?
`warning C4905: wide string literal cast to 'LPCTSTR'`

Reply:
Boooo. Get rid of that type. Please convert all instances of `LPCTSTR` into `LPCWSTR`. The former is the TCHAR logic that always confuses people.

Reply:
@AaronRobinsonMSFT Thanks for your suggestion, this works